code/modules/mob/living/simple_animal/abnormality/waw/hatred_queen.dm

```diff
@@ -276,12 +333,46 @@
 	animate(src, alpha = 255, time = 5)
 	new /obj/effect/temp_visual/guardian/phase/out(teleport_target)
 	forceMove(teleport_target)
+	if(hostile_breach)
+		TeleportExplode()
+
+/mob/living/simple_animal/hostile/abnormality/hatred_queen/proc/ForcedTeleport() //HP teleport for hostile, no checks
```

Comment:
Absolutely doesn't have to be its own proc, just add an argument to the normal TryTeleport and work from that.